src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -498,78 +491,86 @@ public void setMaximumNoDamageTicks(int ticks)
 	@Override
 	public double getLastDamage()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(e != null){
+			return lastEntityDamageEvent.getDamage();
+		}
+		return 0.0;
 	}
 	
 	@Override
 	public void setLastDamage(double damage)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		lastEntityDamageEvent.setDamage(damage);
 	}
-	
+	long lastTimeDamage;
 	@Override
 	public int getNoDamageTicks()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return (int) ((System.currentTimeMillis()-lastTimeDamage)/1000)/20;
 	}
 	
 	@Override
 	public void setNoDamageTicks(int ticks)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		lastTimeDamage = System.currentTimeMillis()-ticks;
```

Comment:
```suggestion
		lastTimeDamage = System.currentTimeMillis()-ticks*(1000/20);
```
Every tick cost 1000/20 = 50 ms